### PR TITLE
Brightness control

### LINKED ISSUE #49: Brightness control
## Description

Provide a way for the user to control the screen's brightness via a slider inside the player controls view.

## Acceptance criteria

- [x] When the user taps to show the controls, the brightness slider is displayed on the left-hand side
- [x] As the user changes the slider level, brightness is changed instantly as visual feedback to the user
- [x] When the user starts interacting with the slider, the rest of the controls are hidden
- [x] After setting the brightness, the slides fades out

## Visual reference

brightness-netflix

### PR DESCRIPTION
**Pull request by thalysmcarrara**
## Description

* Enable user to change activity brightness when fullScreen is enabled

## Related Issues

- Closes #49 

### Pull request checklist

- [ ] **Tests**: This PR includes tests for covering the features or bug fixes (if applicable).
- [ ] **Docs**: This PR updates/creates the necessary documentation.
- [ ] **CI**: Make sure your Pull Request passes all CI checks. If not, clarify the motif behind that and the action plan to solve it (may reference a ticket)

### CODE REVIEW

File: androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/playerOverlay/PlayerControlsScaffold.kt [hunk 1]
```diff
@@ -31,9 +32,9 @@ fun PlayerControlsScaffold(
             modifier = Modifier.fillMaxSize(),
             verticalArrangement = Arrangement.SpaceBetween
         ) {
-            topContent()
-            content()
-            bottomContent()
+            topContent(Modifier.weight(0.2f))
+            content(Modifier.weight(if (isFullScreen) 0.7f else 0.35f))
+            bottomContent(Modifier.weight(if (isFullScreen) 0.3f else 0.45f))
```
Comment (thalysmcarrara) [inline androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/playerOverlay/PlayerControlsScaffold.kt:37]: I forgot those magic numbers, fixed!

File: androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/EnhancedVideoPlayer.kt [hunk 2]
```diff
@@ -78,14 +82,39 @@ fun EnhancedVideoPlayer(
...
     }
 
+    LaunchedEffect(enableImmersiveMode, isFullScreen) {
+        if (enableImmersiveMode) {
+            val shouldShowSystemUi = !isFullScreen
+            context.setStatusBarVisibility(shouldShowSystemUi)
+            context.setNavigationBarVisibility(shouldShowSystemUi)
+        }
+    }
+
```
Comment (aelnonym) [inline androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/EnhancedVideoPlayer.kt:116]: nit: extra \n

File: androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/utils/Extensions.kt [hunk 3]
```diff
@@ -58,3 +60,30 @@ fun Context.setNavigationBarVisibility(showNavigationBar: Boolean) {
...
+        layout?.screenBrightness = BRIGHTNESS_OVERRIDE_NONE
+        layout?.buttonBrightness = BRIGHTNESS_OVERRIDE_NONE
+        it.attributes = layout
+    }
+}
+
+fun Context.getActivityBrightnessOrDefault(default: Float): Float {
+    val window = this.findActivity()?.window
+    var brightness = default
+    window?.let { brightness = it.attributes.screenBrightness }
```
Comment (ricardodalarme) [inline androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/utils/Extensions.kt:87]:
```suggestion
    val brightness = window?.attributes?.screenBrightness ?: default
```

File: androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/utils/Extensions.kt [hunk 4]
```diff
@@ -58,3 +60,30 @@ fun Context.setNavigationBarVisibility(showNavigationBar: Boolean) {
...
+
+    window?.let {
+        val layout: WindowManager.LayoutParams? = it.attributes
+        layout?.screenBrightness = BRIGHTNESS_OVERRIDE_NONE
+        layout?.buttonBrightness = BRIGHTNESS_OVERRIDE_NONE
+        it.attributes = layout
+    }
+}
+
+fun Context.getActivityBrightnessOrDefault(default: Float): Float {
```
Comment (ricardodalarme) [inline androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/utils/Extensions.kt:84]: IMO: it makes more sense to have the default here

```suggestion
fun Context.getActivityBrightnessOrDefault(default: Float = MED_BRIGHTNESS): Float {
```

File: androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/playerOverlay/BrightnessControl.kt [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            brightnessIconState = mapBrightnessRangeToState(brightness)
+            enableHasSliderInteraction()
+        }
+
+        Column(
+            modifier = modifier.fillMaxHeight(),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            brightnessIcon.invoke()
```
Comment (ricardodalarme) [inline androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/playerOverlay/BrightnessControl.kt:80]: nit: usually invoked is used when the variable holding the function is nullable

```suggestion
            brightnessIcon()
```

File: androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/playerOverlay/VerticalSlider.kt [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    modifier: Modifier = Modifier,
+    initialValue: Float,
+    onValueChange: (value: Float) -> Unit,
+    onValueChangeFinished: (() -> Unit)? = null
+) {
+    var value by remember { mutableStateOf(initialValue) }
+
+    Slider(
+        modifier = modifier
+            .rotate(270f).width(DEFAULT_HEIGHT),
```
Comment (ricardodalarme) [inline androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/playerOverlay/VerticalSlider.kt:27]: move the 270f to a constant

File: androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/EnhancedVideoPlayer.kt [hunk 7]
```diff
@@ -1,6 +1,7 @@
 package com.profusion.androidenhancedvideoplayer.components
 
 import android.content.res.Configuration
+import android.util.Log
```
Comment (aelnonym) [inline androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/EnhancedVideoPlayer.kt:4]: nit: not being used

**Review by aelnonym — APPROVED** [review]
Looks good!

**Review by ricardodalarme — APPROVED** [review]

**Pull request merged**